iModelCore/iModelPlatform/DgnCore/DgnDb.cpp

```diff
@@ -665,100 +645,93 @@ DgnDbPtr DgnDb::OpenDgnDb(DbResult* outResult, BeFileNameCR fileName, OpenParams
     dgnDb = new DgnDb(); // release old and create a new DgnDb
     OpenParams readonlyParams(openParams);
     readonlyParams.SetOpenMode(Db::OpenMode::Readonly);
-    status = dgnDb->DoOpenDgnDb(fileName, readonlyParams);
+    status = dgnDb->DoOpenIModel(fileName, readonlyParams);
     return (status != BE_SQLITE_OK) ? nullptr : dgnDb;
-    }
-
-/*---------------------------------------------------------------------------------**//**
-* @bsimethod
-+---------------+---------------+---------------+---------------+---------------+------*/
-DbResult DgnDb::CreateNewDgnDb(BeFileNameCR inFileName, CreateDgnDbParams const& params)
-    {
-    BeFileName projectFile(inFileName);
+}
 
-    if (inFileName.IsEmpty())
-        {
-        projectFile.SetNameUtf8(BEDB_MemoryDb);
-        }
-    else
-        {
-        projectFile.SupplyDefaultNameParts(s_dgndbExt);
-        if (params.m_overwriteExisting && BeFileName::DoesPathExist(projectFile))
-            {
-            if (BeFileNameStatus::Success != BeFileName::BeDeleteFile(projectFile))
-                {
-                LOG.errorv("Unable to create DgnDb because '%s' cannot be deleted.", projectFile.GetNameUtf8().c_str());
+/*---------------------------------------------------------------------------------**/ /**
+ * @bsimethod
+ +---------------+---------------+---------------+---------------+---------------+------*/
+DbResult DgnDb::CreateNewIModel(BeFileNameCR inFileName, CreateDgnDbParams const& params) {
+    BeFileName iModelName(inFileName);
+
+    bool memoryDb = false;
+    if (inFileName.IsEmpty()) {
+        memoryDb = true;
+        iModelName.SetNameUtf8(BEDB_MemoryDb);
+    } else {
+        iModelName.SupplyDefaultNameParts(s_dotBim);
+        if (params.m_overwriteExisting && BeFileName::DoesPathExist(iModelName)) {
+            if (BeFileNameStatus::Success != BeFileName::BeDeleteFile(iModelName)) {
+                LOG.errorv("Unable to create iModel because '%s' cannot be deleted.", iModelName.GetNameUtf8().c_str());
                 return BE_SQLITE_ERROR_FileExists;
-                }
             }
         }
+    }
 
-    bool useSeedDb = !params.m_seedDb.empty();
-
-    if (useSeedDb)
-        {
-        BeFileNameStatus status = BeFileName::BeCopyFile(params.m_seedDb.c_str(), projectFile);
+    if (!params.m_seedDb.empty()) {
+        BeFileNameStatus status = BeFileName::BeCopyFile(params.m_seedDb.c_str(), iModelName);
         if (BeFileNameStatus::Success != status)
             return BE_SQLITE_ERROR_FileExists;
-        }
+    }
 
-    DbResult rc = CreateNewDb(projectFile, params, params.GetGuid());
+    DbResult rc = CreateNewDb(iModelName, params, params.GetGuid());
     if (BE_SQLITE_OK != rc)
         return rc;
 
-    m_fileName = projectFile.GetNameUtf8();
+    // iModels should always use WAL journal mode
+    if (!memoryDb)

```

Comment:
Is this the only actual functional change? Way too much unrelated rename/formatting diff to sift through.

Reply:
yes, the call to EnableWalMode is new and the object of the PR. Several tests broke because they attempted to copy an open file which doesn't work in WAL mode unless you checkpoint first. That precipitated the renames so I could find all callers of create/open.